src/parser/jobs/whm/modules/Lilies.js

```diff
@@ -0,0 +1,182 @@
+					</Trans>
+				</Fragment>,
+				severity: this._unusedLilies >= UNUSED_LILIES_MAX_MINOR ? SEVERITY.MEDIUM : SEVERITY.MINOR,
+				why: <Fragment>
+					<Trans id="whm.lily-cap.suggestion.why">
+						{<Plural value={this._unusedLilies} one="# lily" other="# lilies" />} lost to lily cap.
+					</Trans>
+				</Fragment>,
+			}))
+		}
+	}
+
+	checkBloodLilies() {
+		const possibleBloodLilies = (this._totalLilies / BLOOD_LILY_GENERATION)
+		console.log('possible = ', possibleBloodLilies, this._totalLilies)
+		this._missedBloodLilies = possibleBloodLilies - this._totalBloodLilies
+		const bloodWarnTarget = (possibleBloodLilies - 1) / possibleBloodLilies * 100
+		const bloodFailTarget = (possibleBloodLilies - WASTED_LILIES_MAX_MEDIUM) / possibleBloodLilies * 100
+
+		this.checklist.add(new TieredRule({
+			name: <Trans id="whm.lily-blood.checklist.name">
+				Spend your Blood Lily
+			</Trans>,
+			description: <Trans id="whm.lily-blood.checklist.content">
+				Never use <ActionLink {...ACTIONS.AFFLATUS_SOLACE} /> or <ActionLink {...ACTIONS.AFFLATUS_RAPTURE} /> when <ActionLink {...ACTIONS.AFFLATUS_MISERY} /> is available.  You lose DPS, and may miss the generation of additional Blood Lilies.
+			</Trans>,
+			tiers: {[bloodWarnTarget]: TARGET.WARN, [bloodFailTarget]: TARGET.FAIL, 100: TARGET.SUCCESS},
+			requirements: [
+				new Requirement({
+					name: <Trans id="whm.lily-blood.checklist.requirement.name">
+						Blood Lilies Generated
+					</Trans>,
+					value: this._totalBloodLilies,
+					target: Math.floor(possibleBloodLilies),
+				}),
+			],
+		}))
+
+		if (this._missedBloodLilies >= 1) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.AFFLATUS_MISERY.icon,
+				content: <Fragment>
+					<Trans id="whm.lily-blood.suggestion.content">
+						Use <ActionLink {...ACTIONS.AFFLATUS_MISERY} /> to avoid wasting blood lily growth.
+					</Trans>
+				</Fragment>,
+				severity: this._missedBloodLilies >= WASTED_LILIES_MAX_MEDIUM ? SEVERITY.MAJOR : SEVERITY.MEDIUM,
+				why: <Fragment>
+					<Trans id="whm.lily-blood.suggestion.why">
+						<Plural value={this._missedBloodLilies} one="# blood lily" other="# blood lilies" /> did not bloom due to early lily use.
+					</Trans>
+				</Fragment>,
+			}))
+		}
+	}
+
+	// Calculates the number of lilies the user should currently have.
+	// Lilies are generated every 30 seconds you are in combat, and cap out at 3.
+	_calculateLilies(event) {
+		if (this._lastGained === 0) { this._lastGained = this.parser.fight.start_time }
+
+		const delta = event.timestamp - this._lastGained
+		const newLilies = Math.floor(delta / LILY_INTERVAL)
+		this._lastGained += newLilies * LILY_INTERVAL
+
+		this._lilies = Math.min(MAX_LILIES, this._lilies + newLilies)
+		this._totalLilies += newLilies
+		if (this._lilies + newLilies > MAX_LILIES) {
+			this._unusedLilies += this._lilies + newLilies - MAX_LILIES
+		}
+	}
```

Comment:
I believe this incorrect handling for lilies.  Based on my observations, and testing against a dummy, the charge timer does not run at all while you are at 3 lilies.  Once you hit 3 lilies, no timer is running until you use a lily, at which point the timer starts at 0 and takes 30 seconds to gain you another lily.

Reply:
I will double check when I get home. If that's true then this is a more prominent issue than I thought. Doesn't change much though since what I am predicting is how many you could have had.

Reply:
@AzariahAeroborn You are correct.  Once at 3 lilies, the timer stops counting entirely.  I still need to track it the way I am (to know how many you _could_ have generated), but I also need to track how many you _actually_ generated, as well, for the purpose of watching your blood lilies.  Thanks for pointing out the discrepency. I'll try and get that fixed tomorrow.

Reply:
@AzariahAeroborn I actually wound up not tracking either nearly so closely.  Since capping isn't a major concern, it's very easy to just figure out whether you capped by lily consumption vs. blood lily consumption. At this point, I'm no longer tracking lily generation at all, as I don't think it's necessary.  The end result is we have a minor suggestion about lily capping, a checklist for blood lily spends (since they're always a DPS gain when you have already generated them), and a suggestion when you waste blood lily generation by spending lilies while you have a blood lily.  This should be a more accurate reflection of what "good" play is like.